gdb/riscv-fbsd-tdep.c

```diff
@@ -202,27 +203,36 @@ riscv_fbsd_sigframe_init (const struct tramp_frame *self,
 {
   struct gdbarch *gdbarch = get_frame_arch (this_frame);
   enum bfd_endian byte_order = gdbarch_byte_order (gdbarch);
-  CORE_ADDR sp = get_frame_register_unsigned (this_frame, RISCV_SP_REGNUM);
+  CORE_ADDR frame_sp = get_frame_sp (this_frame);
+  CORE_ADDR ucontext_offset;
+  const struct regcache_map_entry *gp_regmap;
+  size_t gp_regmap_size;
+  if (riscv_abi_clen (gdbarch) != 0)
+    {
+      ucontext_offset = RISCV_SIGFRAME_UCONTEXT_OFFSET_PURECAP;
+      gp_regmap_size = RISCV_FBSD_NUM_CAPREGS * riscv_abi_clen (gdbarch);
+      gp_regmap = riscv_fbsd_capregmap;
+    }
+  else
+    {
+      ucontext_offset = RISCV_SIGFRAME_UCONTEXT_OFFSET_RV64;
+      gp_regmap_size = RISCV_FBSD_NUM_GREGS * riscv_isa_xlen (gdbarch);
+      gp_regmap = riscv_fbsd_gregmap;
+    }
   CORE_ADDR mcontext_addr
-    = (sp
-       + RISCV_SIGFRAME_UCONTEXT_OFFSET
-       + RISCV_UCONTEXT_MCONTEXT_OFFSET);
+    = (frame_sp + ucontext_offset + RISCV_UCONTEXT_MCONTEXT_OFFSET);
+  trad_frame_set_reg_regmap (this_cache, gp_regmap, mcontext_addr,
+			     gp_regmap_size);
+  CORE_ADDR fpregs_addr = mcontext_addr + gp_regmap_size;
+  CORE_ADDR fp_flags_addr = fpregs_addr + RISCV_FBSD_SIZEOF_FPREGSET;
```

Comment:
Leave these where they were and don't reformat fp_flags_addr, you didn't change it

Reply:
looks like I accidentally clang-formatted some lines I didn't change.